lib/Parser/Parse.cpp

```diff
@@ -1398,15 +1398,52 @@ ParseNodePtr Parser::AddVarDeclNode(IdentPtr pid, ParseNodePtr pnodeFnc)
     return pnode;
 }
 
-ParseNodePtr Parser::CreateModuleImportDeclNode(IdentPtr pid)
+Js::PropertyId Parser::EnsurePropertyId(IdentPtr pid)
 {
-    ParseNodePtr declNode = CreateBlockScopedDeclNode(pid, knopConstDecl);
+    Js::PropertyId propertyId = pid->GetPropertyId();
+    if (propertyId == Js::Constants::NoProperty)
+    {
+        propertyId = m_scriptContext->GetOrAddPropertyIdTracked(pid->Psz(), pid->Cch());
+        pid->SetPropertyId(propertyId);
+    }
+    return propertyId;
+}
 
-    declNode->sxVar.sym->SetIsModuleExportStorage(true);
+ParseNodePtr Parser::CreateModuleImportDeclNode(IdentPtr localName)
+{
+    ParseNodePtr declNode = CreateBlockScopedDeclNode(localName, knopConstDecl);
+    Symbol* sym = declNode->sxVar.sym;
+    
+    sym->SetIsModuleExportStorage(true);
 
     return declNode;
 }
 
+Symbol* Parser::MarkIdentifierReferenceIsModuleExport(IdentPtr localName)
+{
+    PidRefStack* pidRef = this->PushPidRef(localName);
```

Comment:
Cool.

--Paul

From: Taylor Woll [mailto:notifications@github.com]
Sent: Monday, March 7, 2016 3:01 PM
To: Microsoft/ChakraCore <ChakraCore@noreply.github.com>
Cc: Paul Leathers <pleath@microsoft.com>
Subject: Re: [ChakraCore] Implement interpreter support for module import/exports (#457)


In lib/Parser/Parse.cpp<https://github.com/Microsoft/ChakraCore/pull/457#discussion_r55288796>:

>

>      return declNode;

>  }

>

> +Symbol* Parser::MarkIdentifierReferenceIsModuleExport(IdentPtr localName)

> +{

> +    PidRefStack* pidRef = this->PushPidRef(localName);

We probably discussed it but somehow I failed to actually implement it this way. This is a great simplification, though, I love it. Made the change.

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/ChakraCore/pull/457/files#r55288796>.